### PR TITLE
List, Create and Edit Surveys of Tracker Program type and handle repeatable sections for Supranational survey

### PR DESCRIPTION
**Pull request by davidbgomes**
### :pushpin: References

-   **Issue:** Closes [Allow auto generated forms to display tracker program forms with repeatable program stages](https://app.clickup.com/t/862kj4qz7)

### :memo: Implementation
- [x] Show profile (enrollment) section to survey;
- [x] Show program stages grouped by Sections
- [x] Allow for repeatable program stages (clicking on "Add new" button)
- [x] Add new Dropdown Question Type 
- [ ] Read data and show on current surveys
- [ ] Write data 

### :video_camera: Screenshots/Screen capture
image

### CODE REVIEW

File: src/webapp/components/survey/SurveyForm.tsx [hunk 1]
```diff
@@ -58,6 +66,11 @@ export const SurveyForm: React.FC<SurveyFormProps> = props => {
     const formClasses = useFormStyles();
     const snackbar = useSnackbar();
     const { api, currentUser } = useAppContext();
+    const [questionnairesSectionsStatus, setQuestionnairesSectionsStatus] = useState<
```
Comment (9sneha-n) [inline src/webapp/components/survey/SurveyForm.tsx:69]: This needs to be a property of the questionnaire class.

File: src/webapp/components/survey/SurveyForm.tsx [hunk 2]
```diff
@@ -58,6 +66,11 @@ export const SurveyForm: React.FC<SurveyFormProps> = props => {
...
+        QuestionnairesSectionsStatusType[]
+    >([]);
+    const [questionnaireSections, setQuestionnaireSections] =
+        useState<_.Dictionary<QuestionnaireSection[]>>();
```
Comment (9sneha-n) [inline src/webapp/components/survey/SurveyForm.tsx:73]: create a hook for all UI logic - the "how" to render needs to be in custom hooks as per clean architecture.

File: src/webapp/pages/survey/SurveyPage.tsx [hunk 3]
```diff
@@ -16,7 +18,7 @@ export const SurveyPage: React.FC = () => {
 
     return (
         <ContentWrapper>
-            <SurveyFormBreadCrumb formType={formType} id={id} />
+            {currentModule?.name === "PPS" && <SurveyFormBreadCrumb formType={formType} id={id} />}
```
Comment (9sneha-n) [inline src/webapp/pages/survey/SurveyPage.tsx:21]: handle bread crumbs for Prevalence survey page

**Comment by davidbgomes** [comment]
Thank you @9sneha-n ! Great work and thank you for stepping in on the PR, just a comment on the edit part of the surveys, it seems to be creating new ones instead of updating, but other than that it looks great!

**Comment by 9sneha-n** [comment]
> Thank you @9sneha-n ! Great work and thank you for stepping in on the PR, just a comment on the edit part of the surveys, it seems to be creating new ones instead of updating, but other than that it looks great!

@davidbgomes Thanks for the review. Good catch!  I will fix "editing existing survey adds new survey for tracker program" in my next PR along with the rest of the prevalence changes.

**Pull request merged**